app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,52 @@
+class CategoriesController < ApplicationController
+
+  def index
+    @categories = Category.all
+    @category = Category.new
+  end
+
+  def show
+    @category = Category.find(params[:id])
+  end
+
+  def new
+    @category = Category.new
+  end
+
+  def create
+    @category = Category.new(category_params)
+    if @category.save
+      flash[:info] = "New category is created"
+      redirect_to '/categories'
+    else
+      redirect_to '/categories'
```

Comment:
@Kohei-Oki Currently, error messages are not displaying in categories. Replace this line with `render 'new'`.